modules/mysql/mysql.go

```diff
@@ -0,0 +1,575 @@
+	   [handler_savepoint_rollback] = [0]
+	   [handler_write] = [5222]
+	   [innodb_buffer_pool_pages_free] = [249]
+	   [innodb_buffer_pool_write_requests] = [1945]
+	   [sort_range] = [0]
+	   [threads_connected] = [3]
+	   [created_tmp_tables] = [20]
+	   [handler_read_first] = [39]
+	   [innodb_buffer_pool_read_requests] = [21518]
+	   [key_blocks_used] = [0]
+	   [max_used_connections] = [3]
+	   [select_full_join] = [2]
+	   [innodb_buffer_pool_pages_flushed] = [215]
+	   [binlog_cache_use] = [0]
+	   [com_insert] = [0]
+	   [com_replace] = [0]
+	   [connection_errors_internal] = [0]
+	   [connection_errors_tcpwrap] = [0]
+	   [innodb_buffer_pool_pages_data] = [257]
+	   [innodb_buffer_pool_pages_dirty] = [0]
+	   [innodb_os_log_written] = [89600]
+	   [threads_created] = [3]
+	   [innodb_data_fsyncs] = [105]
+	   [innodb_data_reads] = [987]
+	   [innodb_rows_read] = [7208]
+	   [key_write_requests] = [0]
+	   [key_writes] = [0]
+	   [innodb_data_pending_writes] = [0]
+	   [innodb_os_log_pending_fsyncs] = [0]
+	   [aborted_connects] = [1]
+	   [handler_prepare] = [0]
+	   [innodb_log_writes] = [98]
+	   [innodb_rows_deleted] = [0]
+	   [innodb_rows_updated] = [346]
+	   [key_blocks_not_flushed] = [0]
+	   [binlog_cache_disk_use] = [0]
+	   [connection_errors_accept] = [0]
+	   [connection_errors_max_connections] = [0]
+	   [connection_errors_select] = [0]
+	   [innodb_buffer_pool_read_ahead_evicted] = [0]
+	   [innodb_os_log_pending_writes] = [0]
+	   [binlog_stmt_cache_use] = [0]
+	   [com_update] = [0]
+	   [handler_delete] = [0]
+	   [handler_read_key] = [2363]
+	   [handler_read_rnd] = [360]
+	   [handler_read_rnd_next] = [10166]
+	   [innodb_data_pending_reads] = [0]
+	   [innodb_rows_inserted] = [0]
+	   [handler_commit] = [678]
+	   [innodb_log_write_requests] = [765]
+	   [opened_files] = [28]
+	   [questions] = [37]
+	   [select_range_check] = [0]
+	   [binlog_stmt_cache_disk_use] = [0]
+	   [innodb_buffer_pool_bytes_dirty] = [0]
+	   [innodb_buffer_pool_reads] = [950]
+	   [innodb_row_lock_current_waits] = [0]
+	   [select_scan] = [35]
+	   [table_locks_immediate] = [12]
+	   [threads_running] = [2]
+	*/
+}
+
+func slaveSeconds(value interface{}) int64 {
+	v, err := strconv.ParseInt(fmt.Sprintf("%s", value), 10, 64)
+	if err != nil {
+		return -1
+	}
+	return v
+}
+
+func slaveRunning(value interface{}) int64 {
+	if v, ok := value.(string); ok {
+		if v == "Yes" {
+			return 1
+		}
+
+		return -1
+	}
+
+	if v := fmt.Sprintf("%s", value); v == "Yes" {
+		return 1
+	}
+
+	return -1
+}
+
+var slaveStats = map[string]func(value interface{}) int64{
+	"Seconds_Behind_Master": slaveSeconds,
+	"Slave_SQL_Running":     slaveRunning,
+	"Slave_IO_Running":      slaveRunning,
+}
+
+// hasReplPriv returns if the current user has privileges for the "SHOW SLAVE STATUS" query.
+func (m *MySQL) hasReplPriv() bool {
+	if m.user == "" {
+		return false
+	}
+
+	rows, err := m.db.Query("select User,Super_priv,Repl_client_priv from mysql.user")
```

Comment:
this doesn't work, user has no permissions to execute SELECT queries

netdata suggested

```
create user 'netdata'@'localhost';
grant usage on *.* to 'netdata'@'localhost';
grant replication client on *.* to 'netdata'@'localhost';  # optional
flush privileges;